expression/builtin_time.go

```diff
@@ -1945,6 +1954,46 @@ func (b *builtinTimeSig) evalDuration(row []types.Datum) (res types.Duration, is
 	return res, isNull, errors.Trace(err)
 }
 
+type timeLiteralFunctionClass struct {
+	baseFunctionClass
+}
+
+func (c *timeLiteralFunctionClass) getFunction(ctx context.Context, args []Expression) (builtinFunc, error) {
+	if err := c.verifyArgs(args); err != nil {
+		return nil, errors.Trace(err)
+	}
+	constant, ok := args[0].(*Constant)
```

Comment:
No. The parser would fail if `args[0]` is not a constant.

Reply:
so why we check whether ok is true here?

Reply:
This can be deleted, but I think it's better to keep it as defensive code. It something goes wrong, it's better to panic than return an weird result.